cli-rs/src/main.rs

```diff
@@ -279,37 +279,9 @@ fn compile_rust(
     coprocessors: Option<String>,
     continuations: bool,
 ) -> Result<(), Vec<String>> {
-    let mut runtime = coprocessors_to_runtime(coprocessors, field.clone());
-
-    if continuations {
-        match field {
-            KnownField::BabyBearField => {
-                if runtime.has_submachine("poseidon_bb") {
-                    return Err(vec![
-                "Poseidon continuations mode is chosen automatically and incompatible with the chosen standard Poseidon coprocessor".to_string(),
-            ]);
-                }
-            }
-            KnownField::Mersenne31Field => {
-                if runtime.has_submachine("poseidon_m31") {
-                    return Err(vec![
-                "Poseidon continuations mode is chosen automatically and incompatible with the chosen standard Poseidon coprocessor".to_string(),
-            ]);
-                }
-            }
-            KnownField::GoldilocksField | KnownField::Bn254Field => {
-                if runtime.has_submachine("poseidon_gl") {
-                    return Err(vec![
-                "Poseidon continuations mode is chosen automatically and incompatible with the chosen standard Poseidon coprocessor".to_string(),
-            ]);
-                }
-            }
-        }
-        runtime = runtime.with_poseidon_for_continuations();
-    }
-
-    let options = CompilerOptions::new(field, runtime);
-    powdr_riscv::compile_rust(file_name, options, output_dir, true, continuations, None)
+    let libs = coprocessors_to_options(coprocessors)?;
+    let options = CompilerOptions::new(field, libs, continuations);
+    powdr_riscv::compile_rust(file_name, options, output_dir, true, None)
```

Comment:
Nice! I wonder if it makes sense to include some of these other parameters into `CompilerOptions` and make use of the "builder" pattern?

Reply:
Just to see how it looks
```
CompilerOptions::new(crate, field)
    .output_dir(out)
    .continuations(true)
    .coprocessors(libs) // could be the Vec<String> directly even and parse inside?
    ...
```

Reply:
yep sounds good

Reply:
I tried for a little bit and it gets kinda messy: in the tests for example, the common functions don't really care what the output dir is, and if I just call `new_32()` which is pretty stateless - which I have to to set up different types of tests, I don't know what output dir to use. So if we bundle everything together I'm gonna have to pass the output_dir around and it might get actually more complicated. In fact, I think the only case where it becomes simpler is here in main.rs.

Reply:
I'll leave it out for now to start extracting this part from this PR, we can still modify it in the new PR if needed.